assessments/projects/openkruise/self-assessment.md

```diff
@@ -0,0 +1,563 @@
+
+###### Actors
+
+- `kruise-manager` - Processes user requests and oversees operations
+- `kruise-daemon` - Performs tasks at node level and monitors status of nodes
+- `kublet` - Runs on each node and can register the node with the apiserver
+
+###### Workflow
+
+1.  `kruise-manager` starts to update a pod
+2.  `kruise-manager` updates the changed fields in the pod
+3.  `kubelet` stops the old container
+4.  `kubelet` pulls the image, creates, and starts new container
+5.  `kruise-manager` updates local conditions for `InPlaceUpdate`
+6.  `kubelet` update Pod Status to be ready
+
+##### Declarative configuration
+
+Kubernetes handles Custom Resource Definitions by default. For each component
+we want to add from OpenKruise a CRD in the form of a yaml file is specified
+with a particular kind and different settings pertaining to that service. Since
+handling these definition are part of Kubernetes, they should not be considered
+as an action for OpenKruise.
+
+#### Goals
+
+##### General
+
+- OpenKruise plays a complementary role to Kubernetes
+- Most features work on Kubernetes clusters without any other dependencies
+- Provide **Advanced Workloads** which extend basic, default Workloads
+- **Decoupled Application Management** to manage apps without modifying Workloads
+  - Provide reliable In-Place update for updating existing Pod images
+- **High-availability Protection** provides extended ways of protecting availability
+- OpenKruise simplifies sidecar injection and enables sidecar In-Place update.
+  OpenKruise also enhances the sidecar startup and termination control.
+- Multi-domain Management: This can help you manage applications over nodes
+  with multiple domains, such as different node pools, available zones,
+  architectures (x86 & arm) or node types (kubelet and virtual kubelet)
+
+##### Security
+
+- OpenKruise components should be protected and robust against tampering
+- Authenticating and authorizing access OpenKruise to control plane components
+- Protect the OpenKruise control plane from being compromised
+
+#### Non-goals
+
+##### General
+
+- OpenKruise is **not** a PaaS and it will **not** provide any abilities of PaaS
+- Providing ways of managing containers without Kubernetes
+- Replacement for kubernetes: OpenKruise is not intended as a means to replace
+  kubernetes, rather it is an extension to Kubernetes providing extra features
+  on top.
+- Standardization of deployment strategies: OpenKruise introduces several
+  advanced deployment strategies, it does not aim to standardize these
+  strategies across Kubernetes cluster. Users are free to manipulate the
+  strategy based on their needs.
+
+##### Security
+
+- OpenKruise doesn't provide additional security for Kubernetes
+- No guarantee of security when an attacker has compromised the system
+
+### Self-assessment use
+
+This self-assessment is created by the OpenKruise team to perform an internal
```

Comment:
Was this created by the OpenKruise maintainer team? Adjust accordingly if not. Consider listing the authors and contributors in the first line of the document, immediately following the title.

Reply:
Thanks for catching, I've updated this section slightly. It now more accurately states at the beginning:
> This self-assessment is created by group of Security Pals with help from the maintainers of OpenKruise to perform an analysis of the project's security.